Setting owner field on Account. Creating test.

ceiroa opened the pull request:
# Critical Changes

# Changes
- The owner ID for an automatically-created Account will now be set to match the owner ID for the Contact.

# Issues Closed
Fixes #622. 

# New Metadata

# Deleted Metadata

# Testing Notes

ceiroa commented:
@xinnanlisf ~ The test and the helper classes were outright copied from NPSP. So in *theory* they should have been code-reviewed already.

ceiroa commented:
@juliesnow Mind taking a look at the sentence under "Changes" and letting me know what you think? Thanks!

xinnanlisf commented on src/classes/UTIL_Profile.cls:
```diff
@@ -0,0 +1,99 @@
+*/
+public class UTIL_Profile {
+    /** @description System Administrator Profile Name */
+    public static final String SYSTEM_ADMINISTRATOR = 'System Administrator';
```
Have a question on this if we need to translate. Would that break this utility class?

ceiroa replied:
That's a great question! @kofijohnson, we copied this code directly from NPSP, and it believe you were the one that wrote it. Any thoughts on Xinnan's question?

annejackson commented:
Should we perhaps add to changes what users would see before this fix, and also what users will see after? I am not sure what we have now for the Changes section would be fully clear. Also stating- is there any action needed? If so, what? would be good here too. @ceiroa  @juliesnow  and @xinnanlisf

ceiroa commented:
@xinnanlisf I modified the code. Let me know what you think. Unfortunately there's no "API Name" field on Profile (see https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_objects_profile.htm for available fields). So I used the option listed in https://developer.salesforce.com/forums/?id=906F00000008zDiIAI. I also replaced an existing reference to "System Administrator" that existed in the code.

ceiroa commented:
Assigning this to you for review and merge @annejackson. The only way I can think of for testing is as follows:

- Disable ACCT_IndividualAccounts_TDTM
- Create a Contact without parent Account (a private Contact)
- Set the OwnerID field to a different User
- Re-enable ACCT_IndividualAccounts_TDTM
- Make a change in the Contact that causes ACCT_IndividualAccounts_TDTM to fire
- Verify the Owner of the automatically-created Account is the same one you set for the Contact

annejackson commented:
Testing has passed. Changes notes look good. Note- there is no action needed from users @aligeez  if you want to include that in any release notes in Hub.
